### PR TITLE
#is helper added

### LINKED ISSUE #1406: {{#is}} and {{#eq}} helper
Are you considering adding stache/mustach ```{{#is}}``` helper in the core?

To allow view statement like this

```{{#is someAttr anotherAttr}} {{/is}}```

```{{^is someAttr 'someConstantValue' }} {{/is}}```

Having this is reducing the need of some exessive code to handle the view state, currenty it is done mainlty with {{if#}} helper,

{{#switch}} https://github.com/bitovi/canjs/issues/1264 migth be useful as well.

### PR DESCRIPTION
**Pull request by Alex0007**
Fixes #1406

### CODE REVIEW

File: view/stache/doc/helpers/is.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+@signature `{{#is expr1 expr2}}BLOCK{{/is}}`
+
+Renders the `BLOCK` template within the current template.
+
+@param {can.stache.key} [args] A key that references a value within the current or parent
+context. If the value is a function or can.compute, the function's return value is used.
+
+@param {can.stache} BLOCK A stache template.
+
+@return {String} If the key's value is truthy, the `BLOCK` is rendered with the
```
Comment (justinbmeyer) [inline view/stache/doc/helpers/is.md]: instead of {String} this will return a {DocumentFragment}.

File: view/stache/doc/helpers/is.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+current context and its value is returned; otherwise, an empty string.
+
+@param {can.stache} BLOCK A template that is rendered
```
Comment (justinbmeyer) [inline view/stache/doc/helpers/is.md]: Typically we always have all params before the @return.

File: view/stache/doc/helpers/is.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@function can.stache.helpers.is {{#is expr1 expr2 expr3}}
```
Comment (justinbmeyer) [inline view/stache/doc/helpers/is.md]: if you look at how the other helpers look in the sidebar of:

http://canjs.com/docs/can.stache.html

You'll notice that `{{#is expr1 expr2 expr3}}` doesn't really fit in.  At least, I think expr3 should be removed: `{{#is expr1 expr2}}`.  We could start showing expanded arguments like `{{#is expr...}}`

File: view/stache/doc/helpers/is.md (hunk 2 above)
Comment (justinbmeyer) [inline view/stache/doc/helpers/is.md]: Actually, is this a duplicate?  I think one of the two BLOCK params should be removed.

File: view/stache/doc/helpers/is.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@parent can.stache.htags 12
+
+@signature `{{#is expr1 expr2}}BLOCK{{/is}}`
```
Comment (justinbmeyer) [inline view/stache/doc/helpers/is.md]: On pages like http://canjs.com/docs/can.stache.helpers.helper.html we use "..." to signal unlimited number of arguments.

Please change the signature to:

`{{#is expr...}}BLOCK{{/is}}`

File: view/stache/doc/helpers/is.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+Renders the `BLOCK` template within the current template.
+
+@param {can.stache.key} [args] A key that references a value within the current or parent
```
Comment (justinbmeyer) [inline view/stache/doc/helpers/is.md]: We don't unfortunately have a can.stache.expression that you can link to.  However, I can fill that in.  Can you change this to:

```
@param {can.stache.expression} [expr...] An expression or key that references a value within the current or parent
```

**Comment by justinbmeyer** [comment]
Looks very good.  I have some documentation suggestions.  Once those are fixed we can get this in.  Thanks!

**Comment by whitecolor** [comment]
@justinbmeyer have looked at the code carefully? it assumes that this helper can take more then two params, is it ok with your vision of it?

**Comment by justinbmeyer** [comment]
Yeah, I'm totally fine with it making sure all arguments are the same.  However, I don't think 2 loops are necessary.  

And, I'm unclear on this part:

```js
  if (can.isFunction(expr)) {
  	curValue = can.compute(expr)();
```

Normally, any function that is passed would be converted to a compute already.  There's no reason to do this afaik.

I would think something like this would work:

```js
var lastValue, curValue;
if(arguments.length - 2 <= 0 ){
  return options.inverse()
}

for (var i = 0; i < arguments.length - 2; i++) {
  var curValue = resolve(arguments[i]);
  if(i >= 0) {
    if(curValue !== lastValue) {
      return options.inverse();
    }
  }
  lastValue = curValue;
 }

return options.fn()
```

**Comment by gsmeets** [comment]
dangit, now I'll have to refactor all my #eq's to #is's. ;)

**Comment by daffl** [comment]
We could alias it. I think I have some `eq`s somewhere, too.

**Comment by whitecolor** [comment]
@justinbmeyer 
Do we need to add ```eq``` alias for this helper to this pull?

**Comment by daffl** [comment]
Thanks! This looks good and will make it into 2.2 with two things:
- Can you squash your commits (http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)
- Let's add the `eq` alias as well.

**Comment by daffl** [comment]
Sweet! Thanks a lot, this is a great feature!

**Comment by gsmeets** [comment]
Good job! And thanks for the alias. That saves me some work.

**Comment by justinbmeyer** [comment]
One thing I noticed is that this isn't using `can.compute.truthy`, so it's possible that there might be unnecessary renderings of something like:

```
{{eq foo.length bar.length}}
```

If foo.length and bar.length were to change together in a batch, and stay the same, this helper would be called twice.

cc @akagomez

**Pull request merged**